protocol/messenger_communities.go

```diff
@@ -1921,3 +1914,389 @@ func (m *Messenger) RequestExtractDiscordChannelsAndCategories(filesToImport []s
 			errors)
 	}()
 }
+
+func (m *Messenger) RequestImportDiscordCommunity(request *requests.ImportDiscordCommunity) {
+	go func() {
+
+		importProgress := &discord.ImportProgress{}
+		importProgress.Init([]discord.ImportTask{
+			discord.CommunityCreationTask,
+			discord.ChannelsCreationTask,
+			discord.ImportMessagesTask,
+			discord.DownloadAssetsTask,
+			discord.InitCommunityTask,
+		})
+		importProgress.CommunityName = request.Name
+
+		exportData, errs := m.ExtractDiscordDataFromImportFiles(request.FilesToImport)
+		if len(errs) > 0 {
+			for _, err := range errs {
+				importProgress.AddTaskError(discord.CommunityCreationTask, err)
+			}
+			m.publishImportProgress(importProgress)
+			return
+		}
+		totalChannelsCount := len(exportData.ExportedData)
+		totalMessageCount := exportData.MessageCount
+
+		if totalChannelsCount == 0 || totalMessageCount == 0 {
+			importError := discord.Error(discord.ErrNoChannelData.Error())
+			if totalMessageCount == 0 {
+				importError.Message = discord.ErrNoMessageData.Error()
+			}
+			importProgress.AddTaskError(discord.CommunityCreationTask, importError)
+			importProgress.StopTask(discord.CommunityCreationTask)
+			m.publishImportProgress(importProgress)
+			return
+		}
+
+		importProgress.UpdateTaskProgress(discord.CommunityCreationTask, 0.5)
+		m.config.messengerSignalsHandler.DiscordCommunityImportProgress(importProgress)
+
+		createCommunityRequest := request.ToCreateCommunityRequest()
+
+		// We're calling `CreateCommunity` on `communitiesManager` directly, instead of
+		// using the `Messenger` API, so we get more control over when we set up filters,
+		// the community is published and data is being synced (we don't want the community
+		// to show up in clients while the import is in progress)
+		discordCommunity, err := m.communitiesManager.CreateCommunity(createCommunityRequest, false)
+		if err != nil {
+			importProgress.AddTaskError(discord.CommunityCreationTask, discord.Error(err.Error()))
+			importProgress.StopTask(discord.CommunityCreationTask)
+			m.publishImportProgress(importProgress)
+			return
+		}
+
+		communitySettings := communities.CommunitySettings{
+			CommunityID:                  discordCommunity.IDString(),
+			HistoryArchiveSupportEnabled: true,
+		}
+		err = m.communitiesManager.SaveCommunitySettings(communitySettings)
+		if err != nil {
+			m.cleanUpFailedImport(discordCommunity)
+			importProgress.AddTaskError(discord.CommunityCreationTask, discord.Error(err.Error()))
+			importProgress.StopTask(discord.CommunityCreationTask)
+			m.publishImportProgress(importProgress)
+			return
+		}
+
+		importProgress.CommunityID = discordCommunity.IDString()
+		importProgress.UpdateTaskProgress(discord.CommunityCreationTask, 0.75)
```

Comment:
Same thing here. `0.75` is arbitrary. Happy to change this. (but also: in reality this happens so fast, that these updates are barely noticable in the UI)